### PR TITLE
subprocpool: populate missing 255 callback arguments

### PR DESCRIPTION
**Pull request by oliver-sanders**
In some situations, these missing arguments could cause tasks to submit fail as the result of a host outage rather than moving onto the next host for the platform.

The bug is as old as the 255 callback - https://github.com/cylc/cylc-flow/pull/4184/files#diff-577155476a8501cbb774dad9dfc50a2ed7b846d577835e58d9bb4dfdb5d56b34R243-R244

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [ ] Tests are included (or explain why tests are not needed).
- [x] Changelog entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Testing these exact lines is rather awkward, however, they are covered.

**Review by wxtim — APPROVED** [review]
Makes sense and fixes a breakage that I introduced all the way back in intelligent host selection (oops). I'd have been happier with it explicitly causing a test failure, but this isn't an easy one to simulate.

Do you want a second review given the lack of tests?

**Review by hjoliver — APPROVED** [review]

**Pull request merged**
